tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/ProcessStart/ProcessStartCommon.cs

```diff
@@ -0,0 +1,89 @@
+// <copyright file="ProcessStartCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Process
+{
+    internal static class ProcessStartCommon
+    {
+        internal const IntegrationId IntegrationId = Configuration.IntegrationId.ProcessStart;
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ProcessStartCommon));
+        internal const string OperationName = "command_execution";
+        internal const string ServiceName = "command";
+        internal const int MaxCommandLineLength = 4096;
+
+        internal static Scope CreateScope(ProcessStartInfo info)
+        {
+            if (info != null)
+            {
+                return CreateScope(info.FileName, info.UseShellExecute ? null : info.EnvironmentVariables);
+            }
+
+            return null;
+        }
+
+        internal static Scope CreateScope(string filename, StringDictionary envVariables)
+        {
+            var tracer = Tracer.Instance;
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId) || !tracer.Settings.IsIntegrationEnabled(IntegrationId.ProcessStart))
+            {
+                // integration disabled, don't create a scope, skip this span
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                Span parent = tracer.InternalActiveScope?.Span;
+
+                if (parent is { Type: SpanTypes.System } && parent.OperationName == OperationName)
+                {
+                    // we are already instrumenting this,
+                    // don't instrument nested methods that belong to the same stacktrace
+                    return null;
+                }
```

Comment:
Is there any scenario for this recursive span check?

Reply:
No, the process.start() method is the only one instrumentated and it should cover all the process.start() overloads. I left this check just as a precaution, but it is truth that we will not need it.